core/client/fs/src/main/java/alluxio/client/file/cache/store/MemoryPageStore.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache.store;
+
+import alluxio.client.file.cache.PageId;
+import alluxio.client.file.cache.PageInfo;
+import alluxio.client.file.cache.PageStore;
+import alluxio.exception.PageNotFoundException;
+import alluxio.exception.status.ResourceExhaustedException;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+
+/**
+ * The {@link MemoryPageStore} is an implementation of {@link PageStore} which
+ * stores all pages in memory.
+ */
+@NotThreadSafe
+public class MemoryPageStore implements PageStore {
+  private static final Logger LOG = LoggerFactory.getLogger(MemoryPageStore.class);
+  private final long mPageSize;
+  private final long mCapacity;
+  private final int mFileBuckets;
+
+  private ConcurrentHashMap<String, byte[]> mPageStoreMap = null;
+
+  /**
+   * Creates a new instance of {@link MemoryPageStore}.
+   *
+   * @param options options for the buffer store
+   */
+  public MemoryPageStore(MemoryPageStoreOptions options) {
+    mPageSize = options.getPageSize();
+    mCapacity = (long) (options.getCacheSize() / (1 + options.getOverheadRatio()));
+    mFileBuckets = options.getFileBuckets();
```

Comment:
do we still need mFileBuckets for mem page store?  I think it's just helpful for file system storage.

Reply:
I will change the policy for the  key

Reply:
Already removed mFileBuckets.